ttnn/cpp/ttnn/operations/pool/generic/device/kernels/dataflow/reader_pool_2d.cpp

```diff
@@ -55,13 +100,7 @@ FORCE_INLINE void read_window_with_top_left_index(uint32_t ind, uint32_t in_l1_r
         uint32_t chunk = 0;
         cb_reserve_back(in_cb_id, 1);
         for (uint32_t h = 0; h < window_h; ++h) {
-            for (uint32_t w = 0; w < window_w; w++) {
-                const uint32_t stick_offset = ind + w + h * in_w_padded;
-                const uint32_t read_offset =
-                    in_l1_read_base_addr + (stick_offset * in_nbytes_c + c_i * MAX_ELE_PER_REDUCTION);
-                noc_async_read_one_packet(get_noc_addr(read_offset), in_l1_write_addr, read_bytes);
-                in_l1_write_addr += in_write_inc;
-                processed_rows++;
+            auto check_row_count = [&]() __attribute__((always_inline)) {
```

Comment:
maybe give up on ternary operators on line 94 and use proper if-else with constexpr where possible